spark-kernel-dsv2/src/main/java/io/delta/spark/dsv2/scan/KernelSparkScanContext.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.spark.dsv2.scan;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.Scan;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.defaults.engine.DefaultEngine;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.spark.dsv2.scan.batch.KernelSparkInputPartition;
+import io.delta.spark.dsv2.utils.SerializableKernelRowWrapper;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.spark.sql.connector.read.InputPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Shared context for Kernel-based Spark scans that manages scan state and partition planning.
+ *
+ * <p>This class is created on the Driver and handles the conversion of Delta Kernel scan files into
+ * Spark InputPartitions. It caches scan file batches to support multiple calls to planPartitions()
+ * and ensures consistent partition planning.
```

Comment:
I commented this elsewhere but commenting here for visibility: let's explicitly include what is the relationship between this class and KernelSparkScan?

Reply:
Or perhaps in relation to KernelSparkBatchScan i think? I defer to you